orchagent/crmorch.cpp

```diff
@@ -336,21 +336,28 @@ void CrmOrch::decCrmAclUsedCounter(CrmResourceType resource, sai_acl_stage_t sta
     {
         m_resourcesMap.at(resource).countersMap[getCrmAclKey(stage, point)].usedCounter--;
 
-        // Remove ACL table related counters
+        // remove acl_entry and acl_counter in this acl table
         if (resource == CrmResourceType::CRM_ACL_TABLE)
         {
-            auto & cntMap = m_resourcesMap.at(CrmResourceType::CRM_ACL_TABLE).countersMap;
-            for (auto it = cntMap.begin(); it != cntMap.end();)
+            for (auto &resourcesMap : m_resourcesMap)
             {
-                if (it->second.id == oid)
+                if ((resourcesMap.first == (CrmResourceType::CRM_ACL_ENTRY))
+                    || (resourcesMap.first == (CrmResourceType::CRM_ACL_COUNTER)))
                 {
-                    it = cntMap.erase(it);
-                }
-                else
-                {
-                    ++it;
+                    auto &cntMap = resourcesMap.second.countersMap;
+                    for (auto it = cntMap.begin(); it != cntMap.end(); ++it)
+                    {
+                        if (it->second.id == oid)
+                        {
+                            cntMap.erase(it);
```

Comment:
Looks like with this change, you are removing ENTRY and COUNTER. Where do we remove the TABLE itself from the `resourcesMap`

Reply:
the resourceMap doesn‘t record acl table, we only need decrement acl_table used count by 1, it was done by the line 338:  m_resourcesMap.at(resource).countersMap[getCrmAclKey(stage, point)].usedCounter--